pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -1332,6 +1332,10 @@ type NodeGroup struct {
 	// The cluster should have been created with all of the local zones specified in this field.
 	// +optional
 	LocalZones []string `json:"localZones,omitempty"`
+
+	// EnclaveEnabled determines if the EC2 instance will be Nitro enclave enabled
+	// +optional
+	EnclaveEnabled bool `json:"enclaveEnabled,omitempty"`
```

Comment:
This should be a pointer so we can differentiate between not provided (nil pointer) and default boolean value of false.